youtube_dl/extractor/rtlnl.py

```diff
@@ -94,19 +94,30 @@ def _real_extract(self, url):
         videopath = material['videopath']
         m3u8_url = meta.get('videohost', 'http://manifest.us.rtl.nl') + videopath
 
-        formats = self._extract_m3u8_formats(m3u8_url, uuid, ext='mp4')
```

Comment:
When downloading a hls videostream, the extension obviously needs to be 'mp4', but initially it really is 'm3u8'.
Can you then think of another way to make a better distinction between the adaptive- and the progressive videostreams? Because right now for the end-user it's quite confusing which formats are adaptive and which are progressive when all extensions are 'mp4'.